src/controller.cpp

```diff
@@ -54,6 +54,10 @@ namespace {
 Logger logger(LOG_CONTROLLER, "Controller");
 
 ControllerImpl::Reason stateToReason(Controller::State state) {
+  if (state == Controller::StateOn) {
```

Comment:
This is an hack :/ Because it's not true that if the state is On we are switching.
Can we debug this more and find a better solution? Or, can you tell us more why you think this is the correct fix?

Reply:
Right, did not think that we need this for deactivation aswell. 
New Approach: 
Have reason be passed explicitly and use `Reason::Switching` for a silent server switch and ```stateToReason()```for all other cases